### PR TITLE
Fix 4593 Permission editor available for unauthorized users

### PR DESCRIPTION
**Pull request by kasongoyo**
## Description
see title

## Issues
 - #4593 
 - ...

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)

 - [x] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)
see #4593

**What is the new behavior?**
Permission editor became unavailable for unauthorized users

**Does this PR introduce a breaking change?** (check one with "x", remove the other)

 - [ ] Yes, and I documented them in migration notes
 - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: web/client/components/resources/modals/Save.jsx [hunk 1]
```diff
@@ -82,6 +84,8 @@ class SaveModal extends React.Component {
      * @return the modal for unsaved changes
     */
     render() {
+        const canEditResourcePermission = this.props.user.role === 'ADMIN' ||
```
Comment (MV88) [inline web/client/components/resources/modals/Save.jsx]:
```suggestion
        const canEditResourcePermission = this.props.user && this.props.user.role === 'ADMIN' ||
```

File: web/client/components/resources/modals/Save.jsx [hunk 2]
```diff
@@ -82,6 +84,8 @@ class SaveModal extends React.Component {
...
+        this.props.resource.attributes && this.props.resource.attributes.owner === this.props.user.name;
```
Comment (MV88) [inline web/client/components/resources/modals/Save.jsx]:
```suggestion
        this.props.resource && this.props.resource.attributes && this.props.resource.attributes.owner === this.props.user.name;
```

File: web/client/components/resources/modals/Save.jsx (hunk 1 above)
Comment (kasongoyo) [reply web/client/components/resources/modals/Save.jsx]: I have initialized user in default prop. @MV88

File: web/client/components/resources/modals/Save.jsx (hunk 2 above)
Comment (kasongoyo) [reply web/client/components/resources/modals/Save.jsx]: Resource is also initialized in default prop.

File: web/client/components/resources/modals/Save.jsx (hunk 1 above)
Comment (MV88) [reply web/client/components/resources/modals/Save.jsx]: user can be null and the default props, works only with undefined values. please update pr

File: web/client/components/resources/modals/Save.jsx (hunk 2 above)
Comment (MV88) [reply web/client/components/resources/modals/Save.jsx]: not sure if resource can be null, but i would do the fix anyway

**Pull request merged**